### PR TITLE
Add support for combined recipe and linux networking version 3

### PR DESCRIPTION
**Pull request by nupuruttarwar**

### CODE REVIEW

**Comment by nupuruttarwar** [comment]
This is a draft PR and not ready for review. This will be ready for review one combined recipe P4 is tested on hardware

**Comment by ffoulkes** [comment]
- DPDK build filed on unidentified symbols in `ovs-p4rt`. The most common cause is adding ES2K-specific code and not conditionalizing it properly.
- clang-format check failed. All new and changed files need to be run through clang-format before being pushed.

**Comment by nupuruttarwar** [comment]
> * DPDK build filed on unidentified symbols in `ovs-p4rt`. The most common cause is adding ES2K-specific code and not conditionalizing it properly.
> * clang-format check failed. All new and changed files need to be run through clang-format before being pushed.

yes Derek, I plan to fix both the issues before I mark the PR as ready for review. It's still in testing phases so it's a draft

**Comment by nupuruttarwar** [comment]
This PR is dependent on https://github.com/ipdk-io/ovs/pull/108, hence the dpdk build fails. I will rebase this PR once OVS PR is merged.

File: ovs-p4rt/ovs_p4rt.cc [hunk 1]
```diff
@@ -306,13 +342,14 @@ void PrepareFdbTableEntryforV4VxlanTunnel(
...
 
+#ifdef LNW_V2
   // Based on p4 program for ES2K, we need to provide a match key SMAC flag
   auto match2 = table_entry->add_match();
   match2->set_field_id(GetMatchFieldId(p4info, L2_FWD_TX_TABLE,
                                        L2_FWD_TX_TABLE_KEY_SMAC_LEARNED));
 
   match2->mutable_exact()->set_value(EncodeByteValue(1, 1));
-
+#endif
```
[the remark this answers is not in the document]
Comment (nupuruttarwar) [reply ovs-p4rt/ovs_p4rt.cc:352]: Derek has suggested to remove the comment if it's a small block

File: CMakeLists.txt [hunk 2]
```diff
@@ -63,6 +63,8 @@ option(IPDK_ROLE_FIX "Enable IPDK role config changes" ON)
 option(SET_RPATH    "Set RPATH in libraries and executables" OFF)
 option(WITH_KRNLMON "Enable Kernel Monitor support" ON)
 option(WITH_OVSP4RT "Enable OVS support" ON)
+option(LNW_V2 "Linux Networking version V2" OFF)
```
Comment (ffoulkes) [inline CMakeLists.txt:66]: Please remove these options. I've modified the LNW_VERSION implementation to define a cmake variable corresponding the variable that is passed to the C/C++ compiler.

File: CMakeLists.txt [hunk 3]
```diff
@@ -77,6 +79,8 @@ endif()
 
 cmake_print_variables(WITH_KRNLMON)
 cmake_print_variables(WITH_OVSP4RT)
+cmake_print_variables(LNW_V2)
```
Comment (ffoulkes) [inline CMakeLists.txt:82]: Please remove these cmake_print_variables() calls. The definitions belong elsewhere, and we really only needed to print them while we were debugging the implementation. (WITH_KRNLMON and WITH_OVSP4RT are considerably more complex, which is why they are being displayed.)

**Review by ffoulkes — COMMENTED** [review]
Please use my implementation of LNW_V2 and LNW_V3 instead of this one. (https://github.com/ipdk-io/krnlmon/pull/112/files)

We do _not_ want to have to specify two different sets of command-line parameters to perform basically the

**Review by ffoulkes — APPROVED** [review]
LGTM

**Review by vsureshkumarp — APPROVED** [review]
LGTM..:)

**Pull request merged**